File: 1281_product_sum_diff.js
var subtractProductAndSum = function(n) {
    let productOfDigits = 1;
    let sumOfDigits = 0;
    while(n > 0) {
        let pulledDigit = n % 10;
        productOfDigits *= pulledDigit;
        sumOfDigits += pulledDigit;
        n = parseInt(n / 10);
    }
    return productOfDigits - sumOfDigits;
};

File: 1281_product_sum_diff.rb
def subtract_product_and_sum(n)
    productOfDigits = 1
    sumOfDigits = 0
    while n > 0 
        pulledDigit = n % 10
        productOfDigits *= pulledDigit
        sumOfDigits += pulledDigit
        n = n / 10 
    end
    return productOfDigits - sumOfDigits;
end